### ISSUE TITLE
Cypress as a node module and multireporters

### ISSUE THREAD

**Issue by Konstruktour** [cypress-io/cypress#1357]
- Operating System: Windows 10
- Cypress Version: 2.02
- Browser Version: Headless

### Is this a Feature or Bug?
Bug

### Current behavior:
Using cypress as a node module and passing a config object including a string with a comma results in unexpected behaviour. String including commas is split up and passed as additional properties. 

### How to reproduce:
Using for eg a multireporter and calling the cypress runner in a nodejs script.

```
cypress.run({
    reporter: 'mocha-multi-reporters',
    reporterOptions: {
        reporterEnabled: 'JSON, spec',
        jsonReporterOptions: {
            toConsole: true
        },
    }
};
```

which calls the run function in lib/cypress.js. 
After line 22
`options = util.normalizeModuleOptions(options);`
the passed options object is transformed to
`reporter: 'mocha-multi-reporters',  
reporterOptions: 'reporterEnabled=JSON, spec,jsonReporterOptions=[object Object]'`

which results in the wrong object (eg at open_project.js, function create)
```
...
     reporter: 'mocha-multi-reporters',
     reporterOptions: 
      { reporterEnabled: 'JSON',
        spec: undefined,
        jsonReporterOptions: 'object Object'
     }
...
```
'spec' is not part of the property reporterEnabled anymore. 
The multireporter cannnot handle this wrong object and runs only one reporter.

**Comment by brian-mann** [comment]
Yup this is a bug. The problem is that all of the args have to be serialized into the command line. We'll have to rethink this approach or update our arg serializer / deserializer.

**Comment by brian-mann** [comment]
Related to: https://github.com/cypress-io/cypress/issues/1348

**Comment by Konstruktour** [comment]
maybe connected: when using a configFile (loaded through plugin as described here https://docs.cypress.io/api/plugins/configuration-api.html#Promises ) which contains nested objects like

```
{
  "reporter": "mocha-multi-reporters",
  "reporterOptions": {
    "reporterEnabled": "mocha-junit-reporter  Spec",
    "mochaJunitReporterReporterOptions": {
      "mochaFile": "./reports/report.xml"
    }
  }
}
```
results in an error 
```
Cannot set property 'mochaFile' of undefined
TypeError: Cannot set property 'mochaFile' of undefined
    at node_modules\cypress\dist\Cypress\resources\app\packages\server\lib\config.js:275:37
```
